### PR TITLE
all: move to v2

### PR DESCRIPTION
**Pull request by rogpeppe**
This is all automated package renaming, except a couple of error message changes caused by using the new versions.

Because of the circular dependency with charmstore, this code depends on an updated branch of charmstore that's yet to land.

### CODE REVIEW

**Comment by jujugui** [comment]
Build finished.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmrepo/347/

File: csclient/csclient_test.go [hunk 1]
```diff
@@ -1875,7 +1875,7 @@ func (s *suite) TestUploadLargeResourceWithHashMismatch(c *gc.C) {
 	}
 	// Upload the resource.
 	_, err = s.client.UploadResource(url, "resname", "data", r, int64(len(r.content)), nil)
-	c.Assert(err, gc.ErrorMatches, `cannot upload part ".*/0": hash mismatch`)
+	c.Assert(err, gc.ErrorMatches, `cannot upload part ".*": hash mismatch`)
```
Comment (urosj) [inline csclient/csclient_test.go]: Why this change?

File: csclient/csclient_test.go [hunk 2]
```diff
@@ -615,7 +615,7 @@ var getArchiveWithBadResponseTests = []struct {
 		Body:          ioutil.NopCloser(strings.NewReader("")),
 		ContentLength: fakeSize,
 	},
-	expectError: `invalid entity id found in response: charm or bundle URL has invalid schema: "no:such"`,
+	expectError: `invalid entity id found in response: cannot parse URL "no:such": schema "no" not valid`,
```
Comment (urosj) [inline csclient/csclient_test.go]: Hm, that's not a mechanical change.

File: csclient/csclient_test.go [hunk 3]
```diff
@@ -761,7 +761,7 @@ func (s *suite) TestUploadArchiveWithServerError(c *gc.C) {
 
 	// Send an invalid hash so that the server returns an error.
 	url := charm.MustParseURL("~charmers/trusty/wordpress")
-	id, err := csclient.UploadArchive(s.client, url, body, hash+"mismatch", size, -1)
+	id, err := csclient.UploadArchive(s.client, url, body, strings.Repeat("0", len(hash)), size, -1)
```
Comment (urosj) [inline csclient/csclient_test.go]: Not a -unstable change?

File: csclient/csclient_test.go (hunk 2 above)
Comment (rogpeppe) [reply csclient/csclient_test.go]: As mentioned in the PR description.

File: csclient/csclient_test.go (hunk 3 above)
Comment (rogpeppe) [reply csclient/csclient_test.go]: Ditto

File: csclient/csclient_test.go (hunk 1 above)
Comment (rogpeppe) [reply csclient/csclient_test.go]: Ditto

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmrepo-merge
